### PR TITLE
Use AppDesigner special file handler in AbstractSpecialFileProvider

### PR DESCRIPTION
**Pull request by davkean**
**Customer scenario**

Opening project properties, choosing the Resources page and choosing _This project does not contain a default resources file. Click here to create one._ fails with:

```
---------------------------
Microsoft Visual Studio
---------------------------
Could not find the file 'c:\users\davkean\documents\visual studio 2017\Projects\ConsoleApp49\ConsoleApp49\Resources.resx'.
---------------------------
OK   
---------------------------
```

**Bugs this fixes:**
https://github.com/dotnet/roslyn-project-system/issues/1078

**Workarounds, if any**
Manually adding a resource file to the project file.

**Risk**
Low.

**Performance impact**
Low

**Is this a regression from a previous update?**
No

**Root cause analysis:**

We didn't miss this - we simply hadn't implemented support, other bugs that were recently fixed prevented us from hitting this path.

Added unit tests - but we still don't have integration tests yet to test the full path.

**How was the bug found?**
Ad hoc testing.

**Developer Notes**
Changed AbstractSpecialFileProvider to call through the AppDesigner special file handler instead of handling/creating itself.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system @srivatsn

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/ResourcesFileSpecialFileProvider.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿    // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/ResourcesFileSpecialFileProvider.cs]: Whitespace

**Comment by basoundr** [comment]
I am surprised there were no test changes here. Don't we have tests for any of these classes?

**Comment by davkean** [comment]
See above:
> Fixing tests to come...

**Comment by basoundr** [comment]
Aha! Missed it

**Comment by davkean** [comment]
tag @MattGertz for review.

**Comment by natidea** [comment]
Shiproom Approved to Merge

**Pull request merged**